src/odata-to-abstract-sql.ts

```diff
@@ -263,11 +269,14 @@ export class OData2AbstractSQL {
 	private extraBodyVars: _.Dictionary<any>;
 	public extraBindVars: ODataBinds;
 	private resourceAliases: _.Dictionary<Resource>;
-	private defaultResource: Resource | undefined;
+	public defaultResource: Resource | undefined;
```

Comment:
`public`?

Reply:
Yeah I need access to that in the `canAccess` callback functions in pinejs, because this holds the current resource.

Reply:
I think it would make more sense to keep it as private and instead do
```ts
return this.methods[prop.method[1].method].call(this, prop);
```
below, along with
```ts
export interface ResourceFunction {
	(this: OData2AbstractSQL, property: any):
		| BooleanTypeNodes
		| { resource: Resource; name: string };
}
```
so that we just run the callback in the same context as the current OData2AbstractSQL instance - that way the code will be most similar to actual in-built code